core/runtime/dynamic_map_internal.odin

```diff
@@ -911,14 +905,12 @@ default_hasher_string :: proc "contextless" (data: rawptr, seed: uintptr) -> uin
 	str := (^[]byte)(data)
 	return default_hasher(raw_data(str^), seed, len(str))
 }
+
 default_hasher_cstring :: proc "contextless" (data: rawptr, seed: uintptr) -> uintptr {
-	h := u64(seed) + INITIAL_HASH_SEED
-	if ptr := (^[^]byte)(data)^; ptr != nil {
-		for ptr[0] != 0 {
-			h = (h ~ u64(ptr[0])) * 0x100000001b3
-			ptr = ptr[1:]
-		}
-	}
-	h &= HASH_MASK
-	return uintptr(h) | uintptr(uintptr(h) == 0)
+	// TODO/perf: len(str) is O(N), and is unavoidable with wyhash.
+	// This is likely insignificant for the typical short key sizes.
```

Comment:
But significant for long key sizes. There was a reason the original hash worked the way it did.